[2025-05-19T07:30:09.232+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-05-19T07:30:09.294+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flight_price_pipeline.load_to_postgres manual__2025-05-19T07:28:22.740128+00:00 [queued]>
[2025-05-19T07:30:09.301+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flight_price_pipeline.load_to_postgres manual__2025-05-19T07:28:22.740128+00:00 [queued]>
[2025-05-19T07:30:09.302+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 3
[2025-05-19T07:30:09.318+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): load_to_postgres> on 2025-05-19 07:28:22.740128+00:00
[2025-05-19T07:30:09.338+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'flight_price_pipeline', 'load_to_postgres', 'manual__2025-05-19T07:28:22.740128+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/flight_price_pipeline.py', '--cfg-path', '/tmp/tmprtt_vwq4']
[2025-05-19T07:30:09.355+0000] {standard_task_runner.py:91} INFO - Job 6: Subtask load_to_postgres
[2025-05-19T07:30:09.352+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:62 DeprecationWarning: This process (pid=233) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-05-19T07:30:09.361+0000] {standard_task_runner.py:64} INFO - Started process 235 to run task
[2025-05-19T07:30:09.436+0000] {task_command.py:426} INFO - Running <TaskInstance: flight_price_pipeline.load_to_postgres manual__2025-05-19T07:28:22.740128+00:00 [running]> on host 286fd7c1dbed
[2025-05-19T07:30:10.358+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='flight_price_pipeline' AIRFLOW_CTX_TASK_ID='load_to_postgres' AIRFLOW_CTX_EXECUTION_DATE='2025-05-19T07:28:22.740128+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-19T07:28:22.740128+00:00'
[2025-05-19T07:30:10.359+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-05-19T07:30:10.378+0000] {loading.py:35} INFO - Starting data loading from MYSQL table flight_prices_raw_transformed to PostgreSQL
[2025-05-19T07:30:15.955+0000] {loading.py:60} INFO - Loaded 57000 rows from MySQL flight_prices_raw_transformed to PostgreSQL
[2025-05-19T07:30:15.959+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-05-19T07:30:15.965+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scripts/loading.py", line 75, in load_to_postgres
    kpi_df = kpis.get(kpi_name)
             ^^^^^^^^
AttributeError: 'str' object has no attribute 'get'
[2025-05-19T07:30:15.993+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=flight_price_pipeline, task_id=load_to_postgres, run_id=manual__2025-05-19T07:28:22.740128+00:00, execution_date=20250519T072822, start_date=20250519T073009, end_date=20250519T073015
[2025-05-19T07:30:16.011+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 6 for task load_to_postgres ('str' object has no attribute 'get'; 235)
[2025-05-19T07:30:16.060+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-05-19T07:30:16.094+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-05-19T07:30:16.095+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
